model/job/broker.go

```diff
@@ -313,61 +314,86 @@ func GetQueuedJobs(db prefixer.Prefixer, workerType string) ([]*Job, error) {
 	return results, nil
 }
 
-// GetJobsBeforeDate returns alls jobs queued before the specified date
-func GetJobsBeforeDate(db prefixer.Prefixer, date time.Time) ([]*Job, error) {
-	var cursor int
-	var jobs []*Job
+func GetAllJobs(db prefixer.Prefixer) ([]*Job, error) {
+	var startkey string
+	var lastJob *Job
 
 	finalJobs := []*Job{}
+	remainingJobs := true
 
-	for cursor != -1 {
-		jobs = []*Job{}
-		req := &couchdb.FindRequest{
-			UseIndex: "by-queued-at",
-			Selector: mango.Lt("queued_at", date.Format(time.RFC3339Nano)),
-			Limit:    1000,
-			Skip:     cursor,
+	for remainingJobs {
+		jobs := []*Job{}
+		req := &couchdb.AllDocsRequest{
+			Limit:    10001,
+			StartKey: startkey,
 		}
-		err := couchdb.FindDocs(db, consts.Jobs, req, &jobs)
+		err := couchdb.GetAllDocs(db, consts.Jobs, req, &jobs)
+
 		if err != nil {
 			return nil, err
 		}
+
+		lastJob, jobs = jobs[len(jobs)-1], jobs[:len(jobs)-1]
```

Comment:
What happens if we have 0 jobs here?

Reply:
I did not think of that case, the commits below should fix it, thanks ;)